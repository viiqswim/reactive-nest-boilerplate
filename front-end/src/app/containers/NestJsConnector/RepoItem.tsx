import React from 'react';
import styled from 'styled-components/macro';

interface Props {
  name?: string;
}

export function RepoItem({ name }: Props) {
  return (
    <Wrapper>
      <Name>{name}</Name>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  padding: 0 1rem;
  min-height: 2.75rem;
  font-weight: 500;
  color: ${p => p.theme.text};

  &:nth-child(odd) {
    background-color: ${p => p.theme.backgroundVariant};
  }
`;

const Name = styled.div`
  flex: 1;
  padding: 0.625rem 0;
`;
