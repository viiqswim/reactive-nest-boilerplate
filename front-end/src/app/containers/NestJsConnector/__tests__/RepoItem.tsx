import React from 'react';
import { render } from '@testing-library/react';
import { RepoItem } from '../RepoItem';

const renderRepoItem = (props: Parameters<typeof RepoItem>[number]) =>
  render(<RepoItem {...props} />);

describe('<RepoItem />', () => {
  it('should match the snapshot', () => {
    const repoItem = renderRepoItem({
      name: 'test',
    });
    expect(repoItem.container.firstChild).toMatchSnapshot();
  });

  it('should have props displayed', () => {
    const name = 'test';
    const repoItem = renderRepoItem({
      name: name,
    });
    expect(repoItem.queryByText(name)).toBeInTheDocument();
  });
});
