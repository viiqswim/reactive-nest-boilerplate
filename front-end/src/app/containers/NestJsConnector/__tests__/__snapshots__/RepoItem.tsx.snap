// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserItem /> should match the snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1rem;
  min-height: 2.75rem;
  font-weight: 500;
}

.c1 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  padding: 0.625rem 0;
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    test
  </div>
</div>
`;
